test/e2e/storage_performant_policy.go

```diff
@@ -0,0 +1,289 @@
+	if err == nil {
+		g.GinkgoLogr.Info("Found configuration for", "host", config.Host)
+	}
+	return config, err
+}
+
+var _ = g.Describe("[sig-storage][OCPFeatureGate:StoragePerformantPolicy] Storage Performant Policy", g.Label("Conformance"), g.Label("Parallel"), func() {
+	var kubeClient *kubernetes.Clientset
+
+	g.BeforeEach(func() {
+		kubeConfig, err := newClientConfigForTest()
+		if err != nil {
+			g.Fail(fmt.Sprintf("Failed to get kubeconfig: %v", err))
+		}
+		kubeClient = kubernetes.NewForConfigOrDie(rest.AddUserAgent(kubeConfig, clientName))
+	})
+
+	g.Context("namespace has fsgroup change policy label", func() {
+		var nsObj *v1.Namespace
+		g.BeforeEach(func(ctx context.Context) {
+			nsObj = &v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "fsgroup-policy-test-",
+					Labels: map[string]string{
+						fsGroupPolicyLabel: string(v1.FSGroupChangeOnRootMismatch),
+					},
+				},
+			}
+			testContext, cancel := context.WithTimeout(ctx, 5*time.Minute)
+			defer cancel()
+
+			var err error
+			nsObj, err = kubeClient.CoreV1().Namespaces().Create(testContext, nsObj, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				g.Fail(fmt.Sprintf("Failed to create test namespace: %v", err))
+			}
+			g.GinkgoLogr.Info("Created namespace with label storage.openshift.io/fsgroup-change-policy=OnRootMismatch", "namespace", nsObj.Name)
+		})
+
+		g.AfterEach(func(ctx context.Context) {
+			testContext, cancel := context.WithTimeout(ctx, 5*time.Minute)
+			defer cancel()
+			cleanupNamespace(testContext, kubeClient, nsObj.Name)
+		})
+
+		g.It("it should set fsgroup change policy to OnRootMismatch if pod has none", func(ctx context.Context) {
+			testContext, cancel := context.WithTimeout(ctx, 5*time.Minute)
+			defer cancel()
+			pod := getPod(nsObj.Name)
+
+			createdPod, err := kubeClient.CoreV1().Pods(nsObj.Name).Create(testContext, pod, metav1.CreateOptions{})
+			if err != nil {
+				g.Fail(fmt.Sprintf("Failed to create test pod: %v", err))
+			}
+			g.GinkgoLogr.Info("Created pod in namespace", "pod", createdPod.Name, "namespace", nsObj.Name)
+
+			// Wait for pod to be scheduled
+			err = wait.PollUntilContextCancel(testContext, waitPollInterval, false, func(ctx context.Context) (bool, error) {
+				p, err := kubeClient.CoreV1().Pods(nsObj.Name).Get(ctx, createdPod.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				return p.Status.Phase != v1.PodPending, nil
+			})
+			if err != nil {
+				g.Fail(fmt.Sprintf("Pod did not schedule: %v", err))
+			}
+
+			// Get the pod and check FSGroupChangePolicy
+			runningPod, err := kubeClient.CoreV1().Pods(nsObj.Name).Get(testContext, createdPod.Name, metav1.GetOptions{})
+			if err != nil {
+				g.Fail(fmt.Sprintf("Failed to get pod: %v", err))
+			}
+			if runningPod.Spec.SecurityContext == nil || runningPod.Spec.SecurityContext.FSGroupChangePolicy == nil {
+				g.Fail("FSGroupChangePolicy not set on pod")
```

Comment:
In general, the error reporting is quite ... short. All that a failed test produces is:

```
fail [github.com/openshift/cluster-storage-operator/test/e2e/storage_performant_policy.go:189]: SELinuxChangePolicy not set on pod
failed: (2m45s) 2025-08-01T10:36:37 "[sig-storage][OCPFeatureGate:StoragePerformantPolicy] Storage Performant Policy namespace has selinux change policy label it should set selinux change policy to Recursive if pod has none"
```

We don't see how the Pod looks like, we don't see any events in the namespace. We don't even see the pod name, so we can't check API server logs.